src/os/filestore/IndexManager.cc

```diff
@@ -116,9 +116,26 @@ int IndexManager::build_index(coll_t c, const char *path, CollectionIndex **inde
   }
 }
 
-int IndexManager::get_index(coll_t c, const string& baseDir, Index *index) {
+bool IndexManager::get_index_optimistic(coll_t c, const string& baseDir, Index *index)
+{
+  RWLock::RLocker l(lock);
 
-  Mutex::Locker l(lock);
+  ceph::unordered_map<coll_t, CollectionIndex* > ::iterator it = col_indices.find(c);
+
+  if (it == col_indices.end()) 
+    return false;
+
+  index->index = it->second;
+
```

Comment:
personally, all of them in this function. because it's a pretty small one. grouping the statements using empty lines does not help much with the readability here.

see https://github.com/ceph/ceph/blob/master/CodingStyle#L21 and https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace